### PR TITLE
Luke/pile refactor

### PR DESCRIPTION
**Pull request by lukesmurray**
I haven't tested this very much. Maybe I could use the testing you built to try to test this. It's a pretty large change and it brought up some issues we might want to take care of.

The general motivation was to make Pile.py a data class. 

- Pile.py no longer maintains logging, since logging and error handling are so coupled this means that Pile.py no longer does error handling. Which raises the question of whether that is a good decision. Maybe Pile.py should do error handling and we should make our logger handle exceptions. This would be more DRY. But would require rewriting some of this pull request
- Pile.py no longer has a test mode. There was never a need for Pile.py to have a test mode except to trigger error logging. Now testing is handled completely in imap.py
- Pile.py no longer has meta methods. Since the entire class is a data class everything is a meta method.
- Pile.py uses the built in [message object](https://docs.python.org/2.7/library/email.message.html#email.message.Message) wherever possible. We no longer get updates in the middle of processing. This could lead to out of sync states but the alternative also had out of sync states but was slower.

### CODE REVIEW

File: smtp_handler/Pile.py [hunk 1]
```diff
@@ -428,7 +216,8 @@ def get_contents(self):
                     decoded = part.get_payload(decode=True)
                     charset = part.get_content_charset()
                     if charset is not None:
-                        decoded = unicode(part.get_payload(decode=True), str(charset), "ignore").encode('utf8', 'replace')
+                        decoded = unicode(part.get_payload(decode=True), str(charset), "ignore").encode('utf8',
+                                                                                                        'replace')
```
Comment (soyapark) [inline smtp_handler/Pile.py]: Could this part be a separate function that handles text encoding and also could be used here: https://github.com/soyapark/murmur/blob/db7a24c4dd9a00dc1bb0c4681e9efbac613e1e49/http_handler/tasks.py#L80-L83

I'd like to have a text encoding handling function that could be used for every user-generated text.

**Comment by soyapark** [comment]
What about Message entity? Are we using both Message and Pile?

**Comment by lukesmurray** [comment]
I think it would be better to switch completely to a `Message` entity. I think I could make the switch rather quickly now that I've done this refactor.

File: smtp_handler/Pile.py (hunk 1 above)
Comment (lukesmurray) [reply smtp_handler/Pile.py]: alright i'll extract this out

**Comment by lukesmurray** [comment]
This is deprecated by #14

**Pull request closed without merging**